firewood/src/shale/mod.rs

```diff
@@ -121,13 +120,13 @@ impl<T: Storable> Obj<T> {
         DiskAddress(NonZeroUsize::new(self.value.get_offset()))
     }
 
-    /// Write to the underlying object. Returns `Ok(())` on success.
+    /// Modifies the value of this object and marks it as dirty.
     #[inline]
-    pub fn write(&mut self, modify: impl FnOnce(&mut T)) -> Result<(), ObjWriteSizeError> {
-        modify(self.value.write());
+    pub fn modify(&mut self, modify_func: impl FnOnce(&mut T)) -> Result<(), ObjWriteSizeError> {
```

Comment:
Renamed because `write` makes it sound like we're writing this object (to disk or a buffer or something) but actually we're not